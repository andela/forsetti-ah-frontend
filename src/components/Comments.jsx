import React from 'react';
import Button from './common/Button';

const Comments = props => (
  <div className='border mb-4 border-left-0 border-right-0 border-bottom-0 col-lg-10'>
    <h6 className='text-uppercase mx-0 mx-md-0 text-left mx-lg-4 pl-0 pb-2 offset-5 mt-2  '>Comments</h6>
    <div className='comment border mx-0 mx-md-0 mx-lg-4  pb-2 shadow-sm'>
      <div className='form-group text-left pl-0 mt-2 mb-0'>
        <textarea
          className='form-control rounded-0  p-2 mb-2 bg-white border-0 shadow-none'
          placeholder='Leave your comments'
          rows='5'
          style={{ resize: 'none' }}
        />
        <div className='row ml-0 col-12'>
          <select className='form-control rounded-0 col-md-3 col-lg-2 col-6 ml-lg-2 ml-1'>
            <option defaultValue='comment'>Comment</option>
            <option value='criticism'>Criticism</option>
          </select>
          <Button className='btn btn_primary ml-2 btn-sm px-3' type='button'> POST </Button>
        </div>
      </div>
    </div>
  </div>
);

export default Comments;
