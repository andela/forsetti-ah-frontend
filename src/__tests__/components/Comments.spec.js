import React from 'react';
import { shallow } from 'enzyme';
import Comments from '../../components/Comments';

const props = {
  commentLoading: false,
  postNewComment: jest.fn(),
  token: 'token',
  history: []
 };
let wrapper;

describe('Comment component', () => {
  test('should render comment component correctly', () => {
    wrapper = shallow(<Comments />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should contain spinner when loading', () => {
    props.commentLoading = true;
    wrapper = shallow(<Comments {...props} />);

    expect(wrapper.exists('Spinner')).toEqual(true);
  });

  test('should correctly set state for user input', () => {
    wrapper = shallow(<Comments {...props} />);
    wrapper.find('textarea').simulate('change', { target: { value: 'comment' } });
    wrapper.find('select').simulate('change', { target: { value: 'criticism' } });

    expect(wrapper.state().comment).toEqual('comment');
    expect(wrapper.state().select).toEqual('criticism');
  });

  test('should post comment on clicking submit button', () => {
    wrapper = shallow(<Comments {...props} />);
    wrapper.find('textarea').simulate('change', { target: { value: 'new comment' } });
    wrapper.find('ButtonComponent').simulate('click');

    expect(props.postNewComment).toHaveBeenCalled();
    expect(wrapper.state().comment).toEqual('');
    expect(wrapper.state().select).toEqual('comment');
  });
});
