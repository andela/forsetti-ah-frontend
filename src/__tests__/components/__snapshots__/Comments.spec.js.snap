// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render comment component correctly 1`] = `ShallowWrapper {}`;
